lib/iris/fileformats/rules.py

```diff
@@ -345,6 +376,34 @@ def _resolve_factory_references(
             aux_factory = factory.factory_class(*args)
             cube.add_aux_factory(aux_factory)
 
+    # In the case of multiple references which vary on a new dimension
+    # (such as time-dependent orography or surface-pressure), the cube may get replaced
+    # by one with a new dimension.
+    # In that case we must update the factory so its dependencies are coords of the
+    # new cube.
+    cube_coord_ids = [
+        id(coord) for coord, _ in cube._dim_coords_and_dims + cube._aux_coords_and_dims
+    ]
+    for factory in cube.aux_factories:
+        for name, dep in list(factory.dependencies.items()):
+            if id(dep) not in cube_coord_ids:
+                factory.update(dep, cube.coord(dep))
+
+    return cube
+
+
+class MultipleReferenceFieldDetector(threading.local):
+    def __init__(self):
+        self.found_multiple_refs = False
+
+
+# A single global object (per thread) to record whether multiple reference fields
+# (e.g. time-dependent orography, or surface pressure fields) have been detected during
+# the latest load operation.
+# This is used purely to implement the iris.LOAD_POLICY.multiref_triggers_concatenate
+# functionality.
+_MULTIREF_DETECTION = MultipleReferenceFieldDetector()
```

Comment:
I don't understand why this would be instantiated at the module level rather than during a load call. Would this not change your loading behaviour depending on what files you have already loaded during your session?

Reply:
It's explained [here](https://github.com/SciTools/iris/blob/e52e20b5c7cba5ebec58e2af7f3029b71d7a4e41/lib/iris/__init__.py#L298-L304), that all the iris.load_xxx load functions call via `_load_collection`, which resets this.  It is admittedly a weak contract (!).
We need it to behave like the settings controls. i.e. to exist as a common (but per-thread) global reference, so it can affect low-level behaviour without the need to pass the control down through all functions in the call chain.